library/cmac.c

```diff
@@ -300,8 +299,10 @@ int mbedtls_cipher_cmac_update( mbedtls_cipher_context_t *ctx,
     /* If there is data left over that wasn't aligned to a block */
     if( ilen > 0 )
     {
-        memcpy( &cmac_ctx->unprocessed_block, input, ilen );
-        cmac_ctx->unprocessed_len = ilen;
+        memcpy( &cmac_ctx->unprocessed_block[cmac_ctx->unprocessed_len],
+                input,
+                ilen );
+        cmac_ctx->unprocessed_len += ilen;
 
         if( ilen % block_size > 0 )
```

Comment:
I think this `if` statement will always evaluate to true. This is because if the value at `ilen` was a multiple of the block size, then if would have been processed in the `for` loop above. I might be wrong though, I am not sure where that call to `mbedtls_cipher_update()` ends up.